tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -193,12 +193,13 @@ def histograms_impl(self, tag, run, downsample_to=None):
         tensor_events = self._multiplexer.Tensors(run, tag)
       except KeyError:
         raise ValueError('No histogram tag %r for run %r' % (tag, run))
+      if downsample_to is not None and len(tensor_events) > downsample_to:
+        rand_indices = random.Random(0).sample(list(range(len(tensor_events))),
```

Comment:
This should pass `six.moves.xrange(len(tensor_events))` to `sample()` instead since it's more efficient: https://docs.python.org/2/library/random.html#random.sample  Right now even on py3 where range() would be efficient, we lose that by converting to list().